users/models/profile.py

```diff
@@ -38,19 +42,24 @@ class Profile(models.Model):
     education = models.CharField(choices=EDUCATION_CHOICES, max_length=50,
                                  null=True)
 
+    @atomic
     def save(self, **kwargs):
         """
         if img is too big we decrease img
         because the less image is the less memory it takes
         """
-        super(Profile, self).save(**kwargs)
-        if self.image:
-            img = Image.open(self.image.path)
-
-            if img.height > 300 or img.width > 300:
-                output_size = (300, 300)
-                img.thumbnail(output_size)
-                img.save(self.image.path)
+        try:
+            super(Profile, self).save(**kwargs)
+            if self.image:
+                img = Image.open(self.image.path)
+                if img.height > 300 or img.width > 300:
+                    output_size = (300, 300)
+                    img.thumbnail(output_size)
+                    img.save(self.image.path)
+        except (OSError, IOError):
+            self.image = None
+            super(Profile, self).save(update_fields=['image'])
+            raise PValidationError('Image can`t be saved')
```

Comment:
Please pay attention to the atomic decorator. An exception means rollback the transaction! I suppose it differs from result you'd may expected